invenio/modules/jsonalchemy/scripts/jsonschema.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Generates JSONSchema output from existing JSONAlchemy models."""
+
+from __future__ import print_function
+
+from invenio.ext.script import Manager
+
+manager = Manager(usage="Generates JSONSchema output from existing JSONAlchemy "
+                  "models.")
+
+@manager.option('-n', '--namespace', dest='namespace', default='recordext',
+                help="Desired namespace")
+@manager.option('-m', '--model', dest='model', help="Desired model. e.g. 'hep'")
+def jsonschema(namespace, model):
+    from json import dumps
+
+    from invenio.modules.jsonalchemy.parser import ModelParser
+    from invenio.modules.jsonalchemy.parser import FieldParser
+
+    from invenio.config import CFG_SITE_URL
+
+    out = {"$schema": "http://json-schema.org/schema#",
+           "id": "%(site_url)s/schemas/%(model)s-0.0.1.json" % {
+               'site_url': CFG_SITE_URL,
+               'model': model
+               },
+           "type": "object",
+           "properties": {},
+           "required": []}
+    fields = ModelParser(namespace).model_definitions(namespace)[model][
+        'fields']
+    out['required'] = fields.keys()
+    for field in fields:
+        parsed_field = FieldParser.field_definition_model_based(field, model,
+                                                                namespace)
+        description = parsed_field.pop('description', None)
+        sub_properties = {}
+        try:
+            for sub_property in parsed_field['producer']['json_for_marc'][0][1].values():
+                sub_properties[sub_property] = {'type': 'string'}
+        except KeyError, IndexError:
+            pass
```

Comment:
Yep, indeed I am trying to improve on this by trying to guess as much as possible and by letting everything that is not parsed into the "other" field. So the user should still have material to complete by hand the conversion...

More improvements will hopefully come this afternoon.